src/traces/splom/index.js

```diff
@@ -183,24 +184,81 @@ function plotOne(gd, cd0) {
         viewOpts.domains[i] = [xa.domain[0], ya.domain[0], xa.domain[1], ya.domain[1]];
     }
 
-    viewOpts.viewport = [gs.l, gs.b, fullLayout.width - gs.r, fullLayout.height - gs.t];
+    viewOpts.viewport = [gs.l, gs.b, gs.w + gs.l, gs.h + gs.b];
 
     if(scene.matrix === true) {
         scene.matrix = createMatrix(regl);
     }
 
+    // FIXME: generate multiple options for single update
     scene.matrix.update(scene.matrixOptions);
     scene.matrix.update(viewOpts);
     scene.matrix.draw();
 }
 
 // TODO splom 'needs' the grid component, register it here?
 
-function hoverPoints() {
+function hoverPoints(pointData, xval, yval, hovermode) {
+    var cd = pointData.cd;
+    var trace = cd[0].trace;
+    var xa = pointData.xa;
+    var ya = pointData.ya;
+    var xpx = xa.c2p(xval);
+    var ypx = ya.c2p(yval);
+    var maxDistance = pointData.distance;
+    var dimensions = trace.dimensions;
+
+    var xi, yi, i;
+    for(i = 0; i < dimensions.length; i++) {
+        if(trace.xaxes[i] === xa._id) xi = i;
+        if(trace.yaxes[i] === ya._id) yi = i;
+    }
+
+    var x = dimensions[xi].values;
+    var y = dimensions[yi].values;
+
+    var id, ptx, pty, dx, dy, dist, dxy;
+    var minDist = maxDistance;
+
+    if(hovermode === 'x') {
```

Comment:
It's a little weird to use compare mode for splom - I guess maybe there could be uses for it, but especially since the same data will appear as both x and y, compare mode will confuse people. Eventually we should default to `closest` when there's a splom, see #778. For now lets just encourage splom users to specify `closest` explicitly.